backend/functions/lib/testing/mock-gateways/mock-oauth2-gateway.ts

```diff
@@ -0,0 +1,73 @@
+import * as jwt from 'jsonwebtoken';
+import * as dotenv from 'dotenv';
+import { randomUUID } from 'crypto';
+import { ApplicationContext } from '../../adapters/types/basic';
+import { ForbiddenError } from '../../common-errors/forbidden-error';
+import { MockRole, usersWithRole } from '../../../../../common/src/cams/mock-role';
+import {
+  CamsJwt,
+  CamsJwtClaims,
+  CamsJwtHeader,
+  OpenIdConnectGateway,
+} from '../../adapters/types/authorization';
+
+dotenv.config();
+
+const MODULE_NAME = 'MOCK_OAUTH2_GATEWAY';
+
+const authIssuer = process.env.AUTH_ISSUER;
+const mockRoles: MockRole[] = usersWithRole;
+const secretKey = randomUUID();
```

Comment:
We need to externalize this secret key. In a cloud environment, different endpoints may be on different machines meaning the secret key is different for each instance. The oauth2 endpoint may use a different key then other endpoints using the mock oidc gateway to verify the access token.

Reply:
...Or just make the secret key === auth issuer.